src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,92 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+
+type Props = {
+  todo: Todo;
+  handleClose: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, handleClose }) => {
```

Comment:
You are initializing your component state with props, which can lead to bugs and make the component harder to understand. It's generally recommended to use props to compute the initial state in the constructor or getDerivedStateFromProps. However, in this case, you don't seem to be using the `todo` prop to initialize any state, so this doesn't seem to be an issue.